src/starknetCli.ts

```diff
@@ -1,58 +1,105 @@
 import assert from 'assert';
 import * as path from 'path';
-import { execSync } from 'child_process';
+import { execSync, ExecSyncOptions } from 'child_process';
 import {
   IDeployProps,
   ICallOrInvokeProps,
   IOptionalNetwork,
   IDeployAccountProps,
   IOptionalDebugInfo,
+  IDeclareOptions,
 } from './index';
 import { encodeInputs } from './passes';
 import { CLIError, logError } from './utils/errors';
-import {
-  getDependencyGraph,
-  hashFilename,
-  reducePath,
-  setDeclaredAddresses,
-} from './utils/postCairoWrite';
+import { getDependencyGraph, hashFilename, reducePath } from './utils/postCairoWrite';
 
 const warpVenvPrefix = `PATH=${path.resolve(__dirname, '..', 'warp_venv', 'bin')}:$PATH`;
+const CAIRO_CMD_DECLARE = ' declare';
+const CAIRO_CMD_STATUS = ' tx_status';
+const CAIRO_CMD_STARKNET_COMPILE = '-compile';
+const CAIRO_CMD_STARKNET_DEPLOY = ' deploy';
+const CAIRO_CMD_STARKNET_DEPLOY_ACC = ' deploy_account';
+const CAIRO_CMD_STARKNET_INVOKE = ' invoke';
+const CAIRO_CMD_STARKNET_CALL = ' call';
 
 interface CompileResult {
   success: boolean;
+  output?: string;
   resultPath?: string;
   abiPath?: string;
   classHash?: string;
 }
 
+function callCairoCommand(cmd: string, command: string, isStdio = true) {
+  const options: ExecSyncOptions = isStdio ? { stdio: 'inherit' } : { encoding: 'utf8' };
+  try {
+    const result = execSync(cmd, options);
+
+    return result;
+  } catch (e) {
+    logError(`StarkNet ${command} failed`);
+  }
+}
+
+function buildCairoCommand(
+  options: any[] | Map<string, string | undefined>,
```

Comment:
Avoid any at all costs.

Reply:
I could change the options type from any [here](https://github.com/NethermindEth/warp/pull/662/commits/0dca7e9f54a2b5d924d14b87095acee55977020b), but for line 47 `multioptions?: any`, I could use some help with this

Reply:
Okay changes made